SpatialAdaptiveMaxPooling.cu

```diff
@@ -0,0 +1,372 @@
+
+#define CUDA_MAX_THREADS 1024   // this is safe, in reality 256 is our limit
```

Comment:
no it's used inside in the kernel, but it's like this in SpatialMaxPooling.cu, SpatialAveragePooling.cu and SubSampling.cu, without undef